File: library/src/main/java/xyz/demj/library/camrecyclerviewadapter/TreeAdapter.java
package xyz.demj.library.camrecyclerviewadapter;

/**
 * Created by demj on 2016/10/16.
 */

public class TreeAdapter<E extends TreeAdapter.TreeNode> extends ConvertAdapter<E> {
    public TreeAdapter(BaseRecyclerViewHolder.ViewHolderFactory<E> pFactory) {
        super(pFactory);
    }

    public interface TreeNode{

    }
}

File: library/src/main/java/xyz/demj/libs/camrecyclerviewadapter/TreeAdapter.kt
package xyz.demj.libs.camrecyclerviewadapter

import android.support.annotation.Keep

/**
 * Created by demj on 2016/10/16.
 */
@Keep
class TreeAdapter<E : TreeAdapter.TreeNode>(pFactory: BaseRecyclerViewHolder.ViewHolderFactory<E>) : ConvertAdapter<E, TreeAdapter.TT<E>>(pFactory, null) {

    interface TreeNode

    class TT<E> : ConvertAdapter.To<E> {

        override fun to(): E? {
            return null
        }
    }
}
